### DISCUSSION TITLE
Blue green deployment

### DISCUSSION THREAD

**Discussion by Arctomachine** [coollabsio/coolify#3460] [Feature Requests]
When new version is deployed, check for incoming connection.
If there are none, just replace container.
If there are, start new version in new container and send new requests there. Keep old container while old connections are active or for specified time frame.

Animation from redhat site about this
![blue-green-deployment-model](https://github.com/user-attachments/assets/946ddf41-896e-4253-bffe-7691198244f5)

**Comment by yes-oo** [comment]
how can coolify help with that any idea ? 
thanks <3

**Reply by Arctomachine** [reply]
Since coolify has access to load balancer and containers, it can in theory easily track old/new connections and decide when to terminate old container.
If done internally, it would be very easy to use for end developers.

But that might be hard task to do manually without native support, not sure how to achieve it meanwhile.

**Comment by acoard** [comment]
Big support for this issue. 

This is also related to the zero-downtime discussion over here. Blue/green deployments are a type of zero-downtime deployments, and while the minimum server requirements would be a bit higher than simply a rolling update, it still gets us zero-downtime deployments. https://github.com/coollabsio/coolify/discussions/3767

**Comment by hades200082** [comment]
Would love to see this with docker compose github deployments

**Comment by paulcsiki** [comment]
Also interested in such a feature.

**Comment by Braunson** [comment]
This is the big feature missing from Coolify, zero downtime deployment.
